sherpa/models/tests/test_model_op.py

```diff
@@ -410,3 +450,245 @@ def test_binop_arithmeticfunction_works(m1, m2):
+                              (abs((a * b) - c), "abs(a * b - c)"),
+                              (abs((a / b) - c), "abs(a / b - c)"),
+                              (a * abs(b * (c + d)), "a * abs(b * (c + d))"),
+                              (abs(b * (c + d)) * (a + d), "abs(b * (c + d)) * (a + d)"),
+                              (-a, "-(a)"),
+                              (+a, "+(a)"),
+                              (-a + b, "-(a) + b"),
+                              (-a + 2, "-(a) + 2.0"),
+                              (+a + b, "+(a) + b"),
+                              (-(a + b), "-(a + b)"),
+                              (-(a * b), "-(a * b)"),
+                              (-(a - b), "-(a - b)"),
+                              (-(a * b - c), "-(a * b - c)"),
+                              (-(a - b * c), "-(a - b * c)"),
+                              (a - a - b, "a - a - b"),
+                              (a - (a - b), "a - (a - b)"),
+                              (a - (b - (c - d)), 'a - (b - (c - d))'),
+                              (a - (b + (c - d)), 'a - (b + c - d)'),
+                              (b - (c + d), 'b - (c + d)'),
+                              ((a - b) - (c + d), 'a - b - (c + d)'),
+                              (a - (b - (c + d)), 'a - (b - (c + d))'),
+                              (a - (b + (c + d)), 'a - (b + c + d)'),
+                              (2 * (a + b) - c * 3, "2.0 * (a + b) - c * 3.0"),
+                              (abs(2 * (a + b) - c * 3), "abs(2.0 * (a + b) - c * 3.0)"),
+                              (a + a, "a + a"),
+                              (a * b, "a * b"),
+                              (a - a, "a - a"),
+                              (a / b, "a / b"),
+                              (a + b + c, "a + b + c"),
+                              (a * b * c, "a * b * c"),
+                              ((a * b) + c, "a * b + c"),
+                              ((a + b) * c, "(a + b) * c"),
+                              (a + (b * c), "a + b * c"),
+                              (a * (b + c), "a * (b + c)"),
+                              ((a + b) * (c + d), "(a + b) * (c + d)"),
+                              ((a * b) * (c + d), "a * b * (c + d)"),
+                              ((a + b) * (c * d), "(a + b) * c * d"),
+                              ((a + (b * c) + d), "a + b * c + d"),
+                              (100 * a * (b + c), "100.0 * a * (b + c)"),
+                              (100 * (a * (b + c)), "100.0 * a * (b + c)"),
+                              (a + b + 2 * c + d + a, "a + b + 2.0 * c + d + a"),
+                              (a + b + c * 2 + d + a, "a + b + c * 2.0 + d + a"),
+                              (a + b * (c - 2) + d + a, "a + b * (c - 2.0) + d + a"),
+                              (a + b * (2 - c) + d + a, "a + b * (2.0 - c) + d + a"),
+                              ((a + b + c) + (c + b + d + a), "a + b + c + c + b + d + a"),
```

Comment:
I'm not asking to change that, I just want to mention that form like this might be chosen by the user for readibility, e.g., the first group is the star and the second group is the accretion disk. I think those cases are rare in practice and that's a drawback I'm happy to accept it. I just want to mention it.

Reply:
Off the top of my head, I thing you'd need to add some sort of explicit constructor to group such elements - e.g.

> group(mdl1 + mdl2 + mdl3) + group(foo + bar * baz)

(but with a name other than `group`), as I don't think you can customize `(` and `)` in Python.

I can see the benefit but, as you mentioned, I am willing to live without it.